app/workers/preneeds/submissions_report.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Preneeds
+  class SubmissionsReport
+    include Sidekiq::Worker
+    def perform(start_date = Time.zone.today - 7, end_date = Time.zone.today)
+      Sentry::TagRainbows.tag
+
+      submissions = PersonalInformationLog.where(error_class: 'PreneedsBurial').where('created_at > ?', start_date)
+                                          .where('created_at < ?', end_date)
+
+      failures = submissions.reject do |sub|
+        sub.decoded_data['response_body'] =~ %r{<returnCode>0</returnCode>}
+      end
+
+      server_unavailable_count = failures.select do |sub|
+        sub.decoded_data['response_body'] =~ /503 Service Unavailable/
+      end.count
+
+      error_persisting_count = failures.select do |sub|
+        sub.decoded_data['response_body'] =~ /Error persisting PreNeedApplication/
+      end.count
+
+      PreneedsSubmissionsReportMailer.build(
+        start_date: start_date.to_s,
+        end_date: (Time.zone.parse(end_date.to_s) - 1).to_date.to_s,
```

Comment:
how about `(end_date - 1.day).to_s`?

Reply:
Since it's a Sidekiq job, the parameters will be stored in Redis as JSON.  So, `start_date` and `end_date` are going to be strings any time this is run asynchronously.  I had to add the `to_s` for times it is run synchronously (e.g. testing)